buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
+								h.AssertNil(t, err)
+
+								h.AssertEq(t, len(br.BOMFiles), 0)
+								expected := "the following SBOM files will be ignored for buildpack api version < 0.7"
+								assertLogEntry(t, logHandler, expected)
+							})
+						})
 					})
 
-					when("buildpack api < 0.7", func() {
-						it("should not include any bom files", func() {
-							bpTOML.API = api.MustParse("0.6").String()
-							buildpackID := bpTOML.Buildpack.ID
-							layerName := "some-layer"
+					when("buildpack api < 0.8", func() {
+						it.Before(func() {
+							h.SkipIf(t, kind == buildpack.KindExtension, "")
+							descriptor.API = "0.7"
+						})
 
-							h.Mkdir(t,
-								filepath.Join(layersDir, buildpackID))
-							h.Mkfile(t, `{"key": "some-bom-content"}`,
-								filepath.Join(layersDir, buildpackID, "launch.sbom.cdx.json"),
-								filepath.Join(layersDir, buildpackID, "build.sbom.cdx.json"),
-								filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.sbom.cdx.json", layerName)))
+						it("does not set environment variables for positional arguments", func() {
+							_, err := descriptor.Build(buildpack.Plan{}, config, mockEnv)
 
-							h.Mkdir(t,
-								filepath.Join(layersDir, buildpackID, layerName))
-							h.Mkfile(t, "[types]\n  cache = true",
-								filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.toml", layerName)))
-
-							br, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
 							h.AssertNil(t, err)
-
-							h.AssertEq(t, len(br.BOMFiles), 0)
-							expected := "the following SBOM files will be ignored for buildpack api version < 0.7"
-							assertLogEntry(t, logHandler, expected)
+							for _, file := range []string{
+								"build-env-cnb-layers-dir-A-v1",
+								"build-env-cnb-platform-dir-A-v1",
+								"build-env-cnb-bp-plan-path-A-v1",
+							} {
+								contents := h.Rdfile(t, filepath.Join(appDir, file))
+								if contents != "unset" {
+									t.Fatalf("Expected %s to be unset; got %s", file, contents)
+								}
+							}
 						})
-					})
-				})
-
-				when("labels", func() {
-					it("should include labels", func() {
-						h.Mkfile(t,
-							"[[labels]]\n"+
-								`key = "some-key"`+"\n"+
-								`value = "some-value"`+"\n"+
-								"[[labels]]\n"+
-								`key = "some-other-key"`+"\n"+
-								`value = "some-other-value"`+"\n",
-							filepath.Join(appDir, "launch-A-v1.toml"),
-						)
-
-						br, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						if err != nil {
-							t.Fatalf("Unexpected error:\n%s\n", err)
-						}
-
-						if s := cmp.Diff(br, buildpack.BuildResult{
-							LaunchBOM: nil,
-							Labels: []buildpack.Label{
-								{Key: "some-key", Value: "some-value"},
-								{Key: "some-other-key", Value: "some-other-value"},
-							},
-							MetRequires: nil,
-							Processes:   []launch.Process{},
-							Slices:      []layers.Slice{},
-						}); s != "" {
-							t.Fatalf("Unexpected:\n%s\n", s)
-						}
-					})
-				})
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L733-L753